src/components/DatePickerRange.react.js

```diff
@@ -0,0 +1,298 @@
+      if(this.state.initialStartDate != newProps.start_date) {
+        this.setState({
+          initialStartDate: newProps.start_date,
+          startDate: momentProps.startDate
+        });
+      }
+      if(this.state.initialEndDate != newProps.end_date) {
+        this.setState({
+          initialEndDate: newProps.end_date,
+          endDate: momentProps.endDate
+        });
+      }
+      if(this.state.firstInitialVisibleMonth != newProps.initial_visible_month){
+        this.setState({
+          firstInitialVisibleMonth: newProps.initial_visible_month,
+          initialVisibleMonth: momentProps.initialVisibleMonth
+        })
+      }
+      if(this.state.initialMinDateRange != newProps.min_date_range){
+        this.setState({
+          initialMinDateRange: newProps.min_date_range,
+          minDateRange: momentProps.minDateRange
+        })
+      }
+      if(this.state.initialMaxDateRange != newProps.max_date_range){
+        this.setState({
+          initialMaxDateRange: newProps.max_date_range,
+          maxDateRange: momentProps.maxDateRange
+        })
+      }
+    }
+
+    render() {
+        const {setProps, fireEvent} = this.props
+        return (
+          <DateRangePicker
+            startDate={ this.state.startDate }
+            startDatePlaceholderText={ this.props.start_date_placeholder_text }
+            endDate={ this.state.endDate }
+            endDatePlaceholderText={ this.props.end_date_placeholder_text }
+            onDatesChange={({ startDate, endDate }) => {
+              this.setState({ startDate, endDate });
+              if (startDate != null) {
+                const startDateStr = startDate.format('YYYY-MM-DD');
+                if (setProps) {
+                  setProps({
+                    startDate: startDateStr
+                  });
+                }
+                if (fireEvent) {
+                  fireEvent('change');
+                }
+              }
+              this.props.start_date = startDate;
+              if (endDate != null) {
+                const endDateStr = endDate.format('YYYY-MM-DD');
+                if (setProps) {
+                  setProps({
+                    endDate: endDateStr
+                  });
+                }
+                if (fireEvent) {
+                  fireEvent('change');
+                }
+              }
+            }}
+            focusedInput={ this.state.focusedInput }
+            onFocusChange={focusedInput => this.setState({ focusedInput })}
+            isOutsideRange={date =>
+              date < this.state.minDateRange || date >= this.state.maxDateRange
+            }
+            showClearDates={ this.props.clearable }
+            disabled={ this.props.disabled }
+            keepOpenOnDateSelect={ this.props.stay_open_on_select }
+            reopenPickerOnClearDates={ this.props.open_calendar_on_clear }
+            initialVisibleMonth={() => {
+              if(this.state.startDate != null) {
```

Comment:
in general, we should use `===` or `!==` wherever possible. Using just `==` or `!=` can lead to some confusing behaviour